src/packages.ts

```diff
@@ -436,39 +434,41 @@ export class RockCXXPackage extends RockPackage
 
 export class RockOrogenPackage extends RockPackage
 {
-    private _bridge: async.EnvironmentBridge;
-
-    constructor(bridge: async.EnvironmentBridge, ws: autoproj.Workspace, info: autoproj.Package, context: context.Context, vscode: wrappers.VSCode)
-    {
-        super(ws, info, context, vscode);
-        this._bridge = bridge;
-    }
-
-    async pickTask(): Promise<async.IOrogenTask | undefined>
+    async pickTask(): Promise<syskit.AvailableDeployment | undefined>
     {
-        let description = this._bridge.describeOrogenProject(this.path, basename(this.path));
+        let syskitConnection = await this.workspace.
+            syskitDefaultConnection(this._vscode);
+        let deployments = syskitConnection.availableDeployments();
         let tokenSource = new vscode.CancellationTokenSource();
 
         let err;
-        description.catch((_err) => {
+        deployments.catch((_err) => {
             err = _err;
             tokenSource.cancel();
         })
-        let promise = description.then(
-            (result) => {
-                let choices: { label: string, description: string, task: async.IOrogenTask }[] = [];
-                result.forEach((task) => {
+        let promise = deployments.then((result) => {
```

Comment:
The point of passing the promise to `showQuickPick` is that you get a cool "loading" animation for free. The longest running operation in this case is `syskitDefaultConnection(this._vscode)`. You should either chain it with `syskitConnection.availableDeployments()` and pass the resulting promise or refactor the first half of this method into an async one and pass the returned promise to `showQuickPick`:

Reply:
... I knew all of this. Just didn't realize I broke it with the await.

Reply:
Sorry for the excessive verbosity then